### PR TITLE
4784/log filter topic type and digest

### LINKED ISSUE #4784: Log Filter Topic type and digest
- [ ] [Log Filter Topic](https://github.com/ChainSafe/web3.js/blob/cc6e15b1de5312b5a665606c1403958073807720/packages/web3-utils/src/types.ts#L25) can also be null or array according to [execution api def](https://github.com/ethereum/execution-apis/blob/68cf607f01bd41aecd0aaafd2bfc01928e188cc2/src/schemas/filter.json#L66) . 
    - [ ] Update [topics](https://github.com/ChainSafe/web3.js/blob/cc6e15b1de5312b5a665606c1403958073807720/packages/web3-common/src/eth_execution_api.ts#L139) type in Log to include `null` and `Topic` (so final type would be `null | Topics | Topics[]`

- [ ] ~Also need to update [topics](https://github.com/ChainSafe/web3.js/blob/cc6e15b1de5312b5a665606c1403958073807720/packages/web3-common/src/eth_execution_api.ts#L139) type in Log to be directly [byte32 array](https://github.com/ethereum/execution-apis/blob/68cf607f01bd41aecd0aaafd2bfc01928e188cc2/src/schemas/receipt.json#L42) instead of `Topics[]` if its updated in above item~

- [ ] Third [param](https://github.com/ChainSafe/web3.js/blob/cc6e15b1de5312b5a665606c1403958073807720/packages/web3-common/src/eth_execution_api.ts#L253) of `eth_submitWork` is digest (also called some time mix hash or mix digest) instead of difficulty [exe api](https://github.com/ethereum/execution-apis/blob/68cf607f01bd41aecd0aaafd2bfc01928e188cc2/src/eth/mining.json#L70).

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

Log Filter Topic type and digest #4784

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-common/src/eth_execution_api.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ export interface Log {
 	readonly blockNumber?: Uint | null;
 	readonly address?: Address;
 	readonly data?: HexStringBytes;
-	readonly topics?: Topic[];
+	readonly topics?: null | Topic | Topic[];
```
Comment (nikoulai) [inline packages/web3-common/src/eth_execution_api.ts:143]: The [issue](https://app.zenhub.com/workspaces/web3js-community-issues-management-612fd4e133c88b001205e29a/issues/chainsafe/web3.js/4784) suggests to : Update [topics](https://github.com/ChainSafe/web3.js/blob/cc6e15b1de5312b5a665606c1403958073807720/packages/web3-common/src/eth_execution_api.ts#L139) type in Log to include null and Topic (so final type would be null | Topics | Topics[]

However, as per [execution api def](https://github.com/ethereum/execution-apis/blob/68cf607f01bd41aecd0aaafd2bfc01928e188cc2/src/schemas/filter.json#L66), [Log Filter Topic](https://github.com/ChainSafe/web3.js/blob/cc6e15b1de5312b5a665606c1403958073807720/packages/web3-utils/src/types.ts#L25) can be bytes32, array of bytes32 or  null. 

So, it wouldn't be better, the initial Topic type to be: HexString32Bytes | HexString32Bytes[] | null and [topics](https://github.com/ChainSafe/web3.js/blob/cc6e15b1de5312b5a665606c1403958073807720/packages/web3-common/src/eth_execution_api.ts#L139) just stay Topic?

Comment (nazarhussain) [reply packages/web3-common/src/eth_execution_api.ts:143]: The `Topic` type is actually synonym of `HexString32Bytes`. So this change is fine.

File: packages/web3-eth/src/rpc_methods.ts [hunk 2]
```diff
@@ -458,13 +458,13 @@ export async function submitWork(
 	requestManager: Web3RequestManager,
 	nonce: HexString8Bytes,
 	seedHash: HexString32Bytes,
```
Comment (nazarhussain) [inline packages/web3-eth/src/rpc_methods.ts]: As per the spec, this parameter is `hash` not `seedHash`. 

https://github.com/ethereum/execution-apis/blob/68cf607f01bd41aecd0aaafd2bfc01928e188cc2/src/eth/mining.json#L63

**Review by nazarhussain — CHANGES_REQUESTED** [review]

**Review by avkos — CHANGES_REQUESTED** [review]
Please fix all things regarding [nazarhussain](https://github.com/nazarhussain) comments.

File: packages/web3-eth/src/rpc_methods.ts (hunk 2 above)
Comment (nikoulai) [reply packages/web3-eth/src/rpc_methods.ts]: fixed

**Review by jdevcs — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Review by nazarhussain — APPROVED** [review]

**Pull request merged**